Sources/TuistKit/Commands/TrackableCommand/TrackableCommand.swift

```diff
@@ -65,54 +51,54 @@ public class TrackableCommand: TrackableParametersDelegate {
     public func run(
         analyticsEnabled: Bool
     ) async throws {
-        let runId: String
         let timer = clock.startTimer()
-        if let command = command as? HasTrackableParameters & ParsableCommand {
-            type(of: command).analyticsDelegate = self
-            runId = command.runId
-            self.command = command
-        } else {
-            runId = UUID().uuidString
-        }
         let pathIndex = commandArguments.firstIndex(of: "--path")
         let path: AbsolutePath
         if let pathIndex, commandArguments.endIndex > pathIndex + 1 {
             path = try AbsolutePath(validating: commandArguments[pathIndex + 1], relativeTo: fileHandler.currentPath)
         } else {
             path = fileHandler.currentPath
         }
-        do {
-            if var asyncCommand = command as? AsyncParsableCommand {
-                try await asyncCommand.run()
-            } else {
-                try command.run()
+        let analyticsStorage = AnalyticsStorage()
```

Comment:
We are currently working with two names "command events" and "analytics" (also reflected in the API). What about renaming this one to:

```suggestion
        let runMetadata = RunMetadataStorage()
```

Or

```suggestion
        let runMetadataRecorder = RunMetadataRecorder()
```